### DISCUSSION TITLE
How can I set headers when using createWSClient?

### DISCUSSION THREAD

**Discussion by samal-rasmussen** [trpc/trpc#2182] [Q&A]
I can pass along a custom header into the context like below. It works for http requests but it doesn't work for the websocket connection. `createWSClient` doesn't take a headers option.

Any ideas?

Client:
```
const wsClient = createWSClient({
  url: `ws://localhost:2022`,
});
const client = createTRPCClient({
  headers: { 'some-custom-header': '42' },
  links: [
    splitLink({
      condition(op) {
        return op.type === 'subscription';
      },
      true: wsLink({
        client: wsClient,
      }),
      false: httpLink({
        url: `http://localhost:2022`,
      }),
    }),
  ],
})
```

Server:
```
export const createContext = async ({
  req,
  res,
}) => {
  return {
    someCustomHeader: req.headers['some-custom-header'],
    req,
    res
  };
};

const { server, listen } = createHTTPServer({
  router: appRouter,
  createContext,
});

const wss = new ws.Server({ server });
applyWSSHandler({
  wss,
  router: appRouter,
  createContext,
});
```

**Comment by jlalmes** [comment]
WebSockets do not support custom headers.
https://stackoverflow.com/questions/4361173/http-headers-in-websockets-client-api

Could you consider using `cookies`?

**Reply by samal-rasmussen** [reply]
Ah, thanks for that info.

I'm not sure how to use cookies with websockets or trpc either. In fact I have never used a cookie before in my life.

I'm trying to build on the [standalone server example](https://github.com/trpc/trpc/tree/main/examples/standalone-server), but I see no use of cookies there either. And also nothing in the [subscriptions docs](https://trpc.io/docs/subscriptions).
